### PR TITLE
Common DTU format, Mac Support

### PR DESCRIPTION
**Pull request by danielbui78**
1. Modifies DazToUnreal to use a common DTU format which will be compatible with Unified Bridge Library
2. Adds Mac Support to DazToUnreal
3. Pre-release version updates for collaboration project

### CODE REVIEW

**Comment by danielbui78** [comment]
### Quality Assurance Review ###

**Common DTU Format update:**
1. Output validation: Produces valid DTU/JSON format. [ PASSED ]
2. Prior Behavior Preserved: Imports asset with multiple materials, assigns correct base type, assigns correct texture filenames, assigns correct alpha values. [ PASSED ]
3. New Feature: DTU/FBX output can be imported with Unity Bridge plugin, producing same result as DazToUnity DS plugin. [ PASSED ]


**Mac support:**
1. Output validation: n/a 
2. Prior Behavior Preserved: Builds and runs correctly on Windows systems. Reads and writes files/folders correctly. [ PASSED ]
3. New Feature: Builds and runs correctly on Mac OS 11.6 with Intel Mac. Reads and writes files/folders correctly. [ PASSED ]


**Added placeholder AssetType property for automation:**
1. Output validation: n/a
2. Prior Behavior Preserved: Able to create and execute DazToUnreal bridge object via Daz Script. [ PASSED ]
3. New Feature: Create DazToUnreal object (DzUnrealAction) and read and write to "AssetType" property. [ PASSED ]


**Pre-release version updates:**
1. Output validation: n/a
2. Prior Behavior Preserved: n/a
3. New Feature: Daz Studio and UE plugin about plugin screens show incremented version number and "Pre-Release" text. [ PASSED ]

File: Unreal/DazStudioPlugin/DzUnrealAction.h
Comment (danielbui78) [inline Unreal/DazStudioPlugin/DzUnrealAction.h]: QString AssetType is unintentionally over-riding QString AssetType from its parent class.  Methods in parent class may not see changes made to AssetType from its child class.

**Background:** 
class DzUnrealAction is a subclass of class DzRuntimePluginAction.  
The DzUnrealAction::executeAction( ) method sets several member variables inherited from its parent class and then invokes the parent class method: DzRuntimePluginAction::Export( ).  This Export( ) method uses the member variables set by DzUnrealAction.

**Solution:**
1. Remove line 24 (declaration of QString AssetType from class DzUnrealAction).
2. This has additional effect that the AssetType property will now be "connected and live", i.e. it will be able to affect bridge export behavior just like modifying the AssetType combobox from the bridge UI.

Comment (danielbui78) [reply Unreal/DazStudioPlugin/DzUnrealAction.h]: **Correction:**
Solution provided above is incomplete, since the value of DzRuntimePluginAction::AssetType is over-written by the value within the GUI combobox for AssetType.  The script-exposed AssetType property can not be directly linked to the GUI combobox value since the GUI is created dynamically and may potentially be NULL at any point in time.  Fully correct solution would be to use signal/slot-based GUI validation system to enforce consistency of DzRuntimePluginAction::AssetType at the time of combobox creation.

Comment (danielbui78) [reply Unreal/DazStudioPlugin/DzUnrealAction.h]: AssetType moved to DzRuntimePluginAction with latest commit.  Script overrides UI and stubs for future GUI to Script integration also implemented.

**Comment by danielbui78** [comment]
Self-Review completed.  I will merge this pull-request once a second person gives approval.

**Review by TotteAlm — APPROVED** [review]

**Pull request merged**
